dask_expr/_merge.py

```diff
@@ -679,6 +679,20 @@ class BlockwiseMerge(Merge, Blockwise):
 
     is_broadcast_join = False
 
+    def _divisions(self):
+        if self.left.npartitions == self.right.npartitions:
+            return super()._divisions()
+        is_unknown = any(d is None for d in super()._divisions())
+        if self.left.npartitions > self.right.npartitions:
+            # Broadcast blockwise merge
+            if is_unknown:
+                return (None,) * (self.left.npartitions + 1)
+            return self.left.divisions
+        else:
+            if is_unknown:
+                return (None,) * (self.right.npartitions + 1)
+            return self.right.divisions
```

Comment:
```suggestion
        frame = self.left if self.left.npartitions > self.right.npartitions else self.right
        if is_unknown:
            return (None,) * (frame.npartitions + 1)
        return frame.divisions
```
nit, take it or leave it. :)